adcpipeline/pipeline.py

```diff
@@ -133,3 +139,37 @@ def run(self) -> None:
         logger.info(f'Running pipeline using the following settings: {self.method_settings}')
         for method in self.method_list:
             method()
+        if self.path is not None:
+            self.df.to_pickle(self.path)
+
+    def run_or_load(self, from_step: int = None) -> None:
+        """
+        Instead of running the pipeline, load the result from the pipeline from a cache file. Useful functionality
+        if you have not modified the pipeline, but want to use the output. Beware that if you change the pipeline, you
+        should re-run the pipeline first or remove the cache. Otherwise this method will load incorrect results.
+        Alternatively you can provide the from_step parameter to load the first N steps from cache and only execute
+        the steps afterwards
+        Args:
+            from_step (Optional): Determine from which step onwards you want to load the cache and continue with
+                executing the pipeline. Each method is accounted as a single step
+
+        """
+        if self.path is None:
+            raise ValueError("Mode not possible without a valid filename attribute in class initiation")
+        if from_step is not None:
+            assert isinstance(from_step, int), "Please provide an int as from_step parameter"
+            step_path = f"{self.path.rstrip('.pkl')}_step{from_step}.pkl"
+            if os.path.isfile(step_path):
+                self.df = pd.read_pickle(step_path)
+                for method in self.method_list[from_step:]:
+                    method()
+            else:
+                for idx, method in enumerate(self.method_list):
+                    if idx == from_step:
+                        self.df.to_pickle(step_path)
```

Comment:
If I want this to happen the first time and later I want to do run the pipeline from, say, step 3, then I think I need to run it as `p.run_or_load(from_step = 0)` and then `p.run_or_load(from_step=3)`, the first line feels a bit odd. I also don't have the option to _not_ cache in the second line. 

Maybe add a `cache_per_step` boolean to use in conjunction with the `from_step`? Then you can just do `p.run_or_load(cache_per_step=True)` and then `p.run_or_load(cache_per_step=False, from_step=3)` if I'm not interested to cache during the second run.